### ISSUE TITLE
Chord with single member not correctly passing results to callback

### ISSUE THREAD

**Issue by bierdybard** [celery/celery#4055]
## Checklist

- [x] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

```
software -> celery:4.0.2 (latentcall) kombu:4.0.2 py:3.5.3
            billiard:3.5.0.2 redis:2.10.5
platform -> system:Linux arch:64bit, ELF imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:redis results:redis://:**@127.0.0.1/3
```

## Steps to reproduce

1. Create two chords with the same callback.
2. In the first chord, call one task in the header. In the second chord, call the same task twice in the header.
3. View the result passed to the chord callback

Demo code:
```
@shared_task
def aa():
    return 'a'

@shared_task
def cb(v):
    logger.debug(v)

@shared_task
def tt():
    chord([aa.s()])(cb.s())
    chord([aa.s(),aa.s()])(cb.s())
```

## Expected behavior
The result passed to the callback is list-wrapped for both callbacks
Expected log output from demo code:
```
[2017-05-26 18:34:08,361: DEBUG/PoolWorker-1] yahoo.contrib.cab_audit.audit.tasks.cb[a6685ec7-017a-4df8-b488-cc1ef8e1bdff]: ['a']
[2017-05-26 18:34:08,382: DEBUG/PoolWorker-5] yahoo.contrib.cab_audit.audit.tasks.cb[b99361d1-f463-4f07-969c-c4d8d193cdfb]: ['a', 'a']
```

## Actual behavior
The result passed to the callback is list-wrapped only when there is more than one task in the header, even if a list is passed in the header.
Actual output from demo code:
```
[2017-05-26 18:34:08,361: DEBUG/PoolWorker-1] yahoo.contrib.cab_audit.audit.tasks.cb[a6685ec7-017a-4df8-b488-cc1ef8e1bdff]: a
[2017-05-26 18:34:08,382: DEBUG/PoolWorker-5] yahoo.contrib.cab_audit.audit.tasks.cb[b99361d1-f463-4f07-969c-c4d8d193cdfb]: ['a', 'a']
```

**Comment by bierdybard** [comment]
It looks like this was caused by the fix for #3323 commited in 867a3aa

**Comment by vnavkal** [comment]
+1, I also observe this bug.  Thanks for reporting it!
